src/transfer/messages.rs

```diff
@@ -68,8 +68,8 @@ pub enum FromClientMessage {
 
 #[derive(Serialize, Deserialize, Debug)]
 pub enum CancelJobResponse {
-    Canceled(Vec<JobTaskId>, JobTaskCount),
-    InvalidJob,
+    Canceled(JobId, Vec<JobTaskId>, JobTaskCount),
```

Comment:
Maybe the `JobId` isn't needed here if the order between `CancelRequest::job_ids` and `CancelJobResponse.0` stays the same?